src/rpc/state_api.rs

```diff
@@ -804,6 +804,60 @@ pub(in crate::rpc) async fn state_vm_circulating_supply_internal<
     )?))
 }
 
+/// Looks back and returns all messages with a matching to or from address, stopping at the given height.
+pub(in crate::rpc) async fn state_list_messages<DB: Blockstore + Send + Sync + 'static>(
+    data: Data<RPCState<DB>>,
+    Params(LotusJson((from_to, tsk, max_height))): Params<
+        LotusJson<(MessageFilter, TipsetKeys, i64)>,
+    >,
+) -> Result<LotusJson<Vec<Cid>>, JsonRpcError> {
+    let ts = data.chain_store.load_required_tipset(&tsk)?;
+
+    if from_to.from == None && from_to.to == None {
+        return Err("must specify at least To or From in message filter".into());
+    } else if let Some(to) = from_to.to {
+        data.state_manager
```

Comment:
Is this call used only for validation? It clearly does not return anything, just propagates an error.

Reply:
Yes, it's only for validation